src/btree/bt_read.c

```diff
@@ -311,7 +311,7 @@ __evict_force_check(WT_SESSION_IMPL *session, WT_PAGE *page)
 	__wt_txn_update_oldest(session, 0);
 
 	/* If eviction cannot succeed, don't try. */
-	return (__wt_page_can_evict(session, page, 1, NULL));
+	return (__wt_page_can_evict(session, page, 1));
```

Comment:
@michaelcahill, @agorrod, OK, I think the light just dawned.

So, the point of this test in the middle of `__wt_page_can_evict` is we explicitly skip the other tests, because in-memory splits don't care about whether or not the page is dirty because we're not really evicting?

I can understand skipping the ` if (btree->checkpointing && (__wt_page_is_modified(page) || F_ISSET(mod, WT_PM_REC_MULTIBLOCK))) ` test, we want to split even if we're checkpointing the tree.

OK, how about moving the "page can split" test FIRST in `__wt_page_can_evict`, and adding a big comment, something like "see if we're going to split, otherwise test for the possibility of actual eviction".